src/components/TodoItem.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { useState, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { updateTodos } from '../api/todos';
+import { Errors } from '../types/Errors';
+
+type Props = {
+  todo: Todo;
+  removeTodo: (id: number) => void;
+  isLoading: boolean;
+  onTodoClick: (id: number, completed: boolean) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  removeTodo,
+  isLoading,
+  onTodoClick,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isSaving, setIsSaving] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleEditSubmit = async () => {
+    if (!editedTitle.trim()) {
+      removeTodo(todo.id);
+    } else if (editedTitle !== todo.title) {
+      setIsSaving(true);
+
+      const updatedTodo = { ...todo, title: editedTitle };
+
+      setEditedTitle(updatedTodo.title);
+      setIsEditing(false);
+
+      try {
+        await updateTodos(updatedTodo);
+        setErrorMessage('');
+      } catch (error) {
+        setErrorMessage(Errors.update);
+      } finally {
+        setIsSaving(false);
+      }
+    } else {
+      setIsEditing(false);
+    }
+  };
+
+  const handleTitleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTitleBlur = () => {
+    handleEditSubmit();
+  };
+
+  useEffect(() => {
+    const handleKeyPress = (event: KeyboardEvent) => {
+      if (event.key === 'Enter') {
+        handleEditSubmit();
+      } else if (event.key === 'Escape') {
+        setEditedTitle(todo.title);
+        setIsEditing(false);
+      }
+    };
+
+    if (isEditing) {
+      window.addEventListener('keydown', handleKeyPress);
+    } else {
+      window.removeEventListener('keydown', handleKeyPress);
+    }
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyPress);
+    };
+  }, [isEditing, editedTitle, todo]);
```

Comment:
In general this is not the best wait to listen for a keyDown events.
Please use standard JSX/React approach with a `onKeyDown` event handler which is put inside JSX element.